src/client/ConnectionPool.cpp

```diff
@@ -30,19 +33,27 @@ void ConnectionPool::init(const std::vector<std::string> &addresses, const Confi
     }
     config_ = config;
     newConnection(0, config.maxConnectionPoolSize_);
+
+    keepAlive_.store(true, std::memory_order_release);
+    maintainer_ = std::thread(&ConnectionPool::keepAliveJob, this);
```

Comment:
why use thread to add and remove the connection, it has a time delay.

Reply:
It's used to reduce the time when retry connection, if don't keep alive of connection pool we need open new connection when retry connection.